Purchases/Networking/Operations/GetIntroEligibilityOperation.swift

```diff
@@ -0,0 +1,133 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  GetIntroEligibilityOperation.swift
+//
+//  Created by Joshua Liebowitz on 11/19/21.
+
+import Foundation
+
+class GetIntroEligibilityOperation: NetworkOperation {
+
+    private let configuration: UserSpecificConfiguration
+    private let receiptData: Data
+    private let productIdentifiers: [String]
+    private let completion: IntroEligibilityResponseHandler
+
+    init(configuration: UserSpecificConfiguration,
+         receiptData: Data,
+         productIdentifiers: [String],
+         completion: @escaping IntroEligibilityResponseHandler) {
+        self.configuration = configuration
+        self.receiptData = receiptData
+        self.productIdentifiers = productIdentifiers
+        self.completion = completion
+
+        super.init(configuration: configuration)
+    }
+
+    override func main() {
+        if self.isCancelled {
+            return
+        }
+
+        self.getIntroEligibility(appUserID: self.configuration.appUserID,
+                                 receiptData: self.receiptData,
+                                 productIdentifiers: self.productIdentifiers,
+                                 completion: self.completion)
+    }
+
+    func getIntroEligibility(appUserID: String,
+                             receiptData: Data,
+                             productIdentifiers: [String],
+                             completion: @escaping IntroEligibilityResponseHandler) {
+        guard productIdentifiers.count > 0 else {
+            completion([:], nil)
+            return
+        }
```

Comment:
Yeah, I think you're right here. This should be an error. It's _probably_ a programming error to get empty productIds.

Reply:
It's not so easy to do this... the `[:]` is there because it's not nullable, and following this down the rabbit hole becomes a bit more of a refactor than I'm willing to do. I want to send `(nil, Error)`, but that breaks a bunch of stuff. So I'm going to not do anything here and leave it as it was. Unfortunately.